lib/groonga-query-log/filter-rewriter.rb

```diff
@@ -77,5 +80,21 @@ def rewrite_nullable_reference_number(filter)
         end
       end
     end
+
+    def rewrite_regular_expression(filter)
+      filter.gsub(
+        /(?<before_expression>.+) *&& *(?<target_column>\w+) *@~ *\"\^\(\?\!\.\*(?<head_value>\w+) *(?:\| *\w+ *)*\)\.\+\$\"/
```

Comment:
We can use `/.../x` for long regular expression:

```ruby
/(?<before_expression>.+)\ *&&
 \ *(?<target_column>\w+)\ *@~\ *
 \"\^\(\?\!\.\*(?<head_value>\w+)\ *(?:\|\ *\w+\ *)*\)\.\+\$\"/x
```

Reply:
Is `(?<before_expression>.+)` needed?

Reply:
`(?<target_column>\w+)` doesn't match `Table.index`.

Reply:
`(?<target_column>\w+)` matches `column2あ`.
We're using `([a-zA-Z0-9_.]+)` for left hand side of expression in this class.
(The regular expression isn't enough because it matches `a...`.)

Reply:
`\` is needless for `\"`.

Reply:
`\^`should be `(?:\^|\\A)`.

Reply:
`\$` should be `(?:\$|\\z)`.

Reply:
`(?<head_value>\w+) *` doesn't match `xxx yyy`.

Reply:
What about this?

```ruby
filter.gsub(/&& *(?<target_column>[a-zA-Z0-9_.]+) *@~ *"(?<pattern>.*?)"/) do |matched|
  target_column = $LAST_MATCH_INFO[:target_column]
  pattern = $LAST_MATCH_INFO[:pattern]

  case pattern
  when /\A(?<header>(?:\^|\\A)\(\?\!\.\*)
          (?<body>.+)
          (?<footer>\)\.[+*](?:\$|\\z))\z/x
    header = $LAST_MATCH_INFO[:header]
    body = $LAST_MATCH_INFO[:body]
    footer = $LAST_MATCH_INFO[:footer]
    conditions = body.split("|").collect do |word|
      "&! #{target_column} @ \"#{word.strip}\""
    end
    conditions.join(" ")
  else
    matched
  end
end
```

NOTE: Backslash escape in string literal isn't supported.